playwright/tests/auth.setup.ts

```diff
@@ -0,0 +1,22 @@
+import { HomePage } from "@pages/home-page";
+import { LoginPage } from "@pages/login-page";
+import {
+  expect,
+  test as setup,
+} from "@playwright/test";
+
+const adminFile = "playwright/.auth/admin.json";
+
+setup("authenticate as admin", async ({ page }) => {
+  const loginPage = new LoginPage(page);
+  await loginPage.goto();
+  await loginPage.typeEmail(process.env.CYPRESS_USER_NAME!);
+  await loginPage.typePassword(process.env.CYPRESS_USER_PASSWORD!);
+  await loginPage.clickSignInButton();
+
+  const homePage = new HomePage(page);
+  await expect(homePage.welcomeMessage).toContainText("Hello there,");
+
+  // End of authentication steps.
+  await page.context().storageState({ path: adminFile });
+});
```

Comment:
Do we need to log in via frontend in setup?

Reply:
This is the main and recommended way by PW https://playwright.dev/docs/auth

Reply:
You do it just once per role before all tests start and then all tests uses same storage and cookies 🍪  - looks as really nice solution

Reply:
Plus you can treat it as sanity test - if you can't log in why bother testing rest of the stuff

Reply:
Authenticate with API request is also recommended. From playwright docs:
When to use

Your web application supports authenticating via API that is easier/faster than interacting with the app UI.

I think in most cases test start to fail when somthing Has changed on the UI, not when it's broken. So in the case of missing data-test-id on some input, all tests will be broken, not just login. You will need to fix it asap, on order to run tests

Reply:
@karola312 I see your point 🤔 but still I think it will not be a problem, it is beneficial since we have a copy of the browser state which is much more correlated to the end user. If this will cause trouble then we need to fix it in one place (setup) so let's give it a shot - if won't do the trick I will change it to API. Is that fair enough?